lib/modules/datasource/python-version/index.spec.ts

```diff
@@ -0,0 +1,147 @@
+      expect(
+        res?.releases.find((release) => release.version === '3.7.17')
+          ?.isDeprecated,
+      ).toBeTrue();
+    });
+  });
+
+  describe('getReleases', () => {
+    beforeEach(() => {
+      httpMock
+        .scope('https://endoflife.date')
+        .get('/api/python.json')
+        .reply(200, Fixtures.get('eol.json'));
+
+      jest.spyOn(githubGraphql, 'queryReleases').mockResolvedValueOnce([
+        {
+          id: 1,
+          url: 'https://example.com',
+          name: 'containerbase/python-prebuild',
+          description: 'some description',
+          version: '3.12.1',
+          releaseTimestamp: '2020-03-09T13:00:00Z',
+        },
+        {
+          id: 2,
+          url: 'https://example.com',
+          name: 'containerbase/python-prebuild',
+          description: 'some description',
+          version: '3.12.0',
+          releaseTimestamp: '2020-03-09T13:00:00Z',
+        },
+        {
+          id: 3,
+          url: 'https://example.com',
+          name: 'containerbase/python-prebuild',
+          description: 'some description',
+          version: '3.7.8',
+          releaseTimestamp: '2020-03-09T13:00:00Z',
+        },
+      ]);
+    });
+
+    it('throws for 500', async () => {
+      httpMock.scope(defaultRegistryUrl).get('').reply(500);
+      await expect(
+        getPkgReleases({
+          datasource,
+          packageName: 'python',
+        }),
+      ).rejects.toThrow(EXTERNAL_HOST_ERROR);
+    });
+
+    it('returns null for error', async () => {
+      httpMock.scope(defaultRegistryUrl).get('').replyWithError('error');
+      expect(
+        await getPkgReleases({
+          datasource,
+          packageName: 'python',
+        }),
+      ).toBeNull();
+    });
+
+    it('returns null for empty 200 OK', async () => {
+      httpMock.scope(defaultRegistryUrl).get('').reply(200, []);
+      expect(
+        await getPkgReleases({
+          datasource,
+          packageName: 'python',
+        }),
+      ).toBeNull();
+    });
+
+    describe('processes real data', () => {
+      beforeEach(() => {
+        httpMock
+          .scope(defaultRegistryUrl)
+          .get('')
+          .reply(200, Fixtures.get('release.json'));
+      });
+
+      it('returns the correct data', async () => {
+        const res = await getPkgReleases({
+          datasource,
+          packageName: 'python',
+        });
+        expect(res?.releases[0]).toEqual({
+          isDeprecated: true,
+          isStable: true,
+          releaseTimestamp: '2020-06-27T12:55:01.000Z',
+          version: '3.7.8',
+        });
+      });
+
+      it('returns no unstable versions', async () => {
+        const res = await getPkgReleases({
+          datasource,
+          packageName: 'python',
+        });
+        res?.releases.forEach((release) => {
+          expect(release.isStable).toBeTrue();
+        });
+      });
+
+      it('returns no version that is not prebuilt', async () => {
```

Comment:
```suggestion
      it('only returns versions that are prebuilt', async () => {
```

Some idea here to focus on the _correct_ condition.

Check if this is correct before applying the suggestion.